### ISSUE TITLE
Environment variables documented as inserted into corefile (but not quite so)

### ISSUE THREAD

**Issue by Kerrnel** [coredns/coredns.io#236]
Putting "1.1.1.1 8.8.8.8 9.9.9.9" into FORWARD variable
Referencing {$FORWARD} in e.g.

`
forward . {$FORWARD}
`

gives an error that it is not an IP or filename.

No such restriction is documented afaics.

**Comment by miekg** [comment]
True. Would assume this just work, but I gather the parser sees the entire
env as one token

**Comment by chrisohaver** [comment]
CoreDNS uses Caddy 1, wherein env vars expand into a single token only. This has changed in Caddy 2.
